packages/orchestration/src/facade.js

```diff
@@ -109,25 +109,16 @@ export const makeOrchestrationFacade = ({
    * @returns {{ [N in keyof GFM]: HostForGuest<GFM[N]> }}
    */
   const orchestrateAll = (guestFns, hostCtx) => {
-    const getMappedFlows = () => {
-      return Object.fromEntries(
-        Object.keys(guestFns).map(name => [
-          name,
-          // eslint-disable-next-line no-use-before-define
-          (...args) => orcFns[name](...args),
-        ]),
-      );
-    };
-
-    const mappedContext = Object.fromEntries(
-      Object.entries(hostCtx).map(([key, value]) => [
-        key,
-        // TODO: support matching individual guest functions anywhere in the context
-        // instead of matching the record as a whole
-        // https://github.com/Agoric/agoric-sdk/issues/9823
-        value === guestFns ? getMappedFlows() : value,
+    const mappedFlows = new Map(
+      Object.entries(guestFns).map(([name, guestFn]) => [
+        guestFn,
+        // eslint-disable-next-line no-use-before-define
+        (...args) => orcFns[name](...args),
       ]),
     );
+    const mapFlow = guestFn => mappedFlows.get(guestFn) || guestFn;
```

Comment:
a little awkward that `mapFlow` uses `mappedFlows`.

consider inlining this fn